relay.go

```diff
@@ -95,18 +85,17 @@ func NewRelay(ctx context.Context, h host.Host, upgrader *tptu.Upgrader, opts ..
 		self:     h.ID(),
 		incoming: make(chan *Conn),
 		relays:   make(map[peer.ID]struct{}),
+		isAllowedToHop: func(_ network.Stream) bool {
+			return false
+		},
```

Comment:
This is not a reasonable default! The default should be to allow all nodes to hop.

Reply:
This needs to be separate from the filter.

Reply:
I completely agree with you but that not the current default and I want to keep this change as non breaking as possible, I will change that if more people agree with us.